### PR TITLE
216 search functionalities

### PR DESCRIPTION
**Pull request by NurElHuda**
# Description

Adds search through the website.
Implements #216

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

### CODE REVIEW

File: frontend/src/apps/main/redux/reducers/articles-page/index.ts [hunk 1]
```diff
@@ -12,6 +12,8 @@ export const articlesPage = (
     payload: ArticlesPageInitialState;
   },
 ) => {
+  console.log(state);
```
Comment (Souhaib-Benbouzid) [inline frontend/src/apps/main/redux/reducers/articles-page/index.ts:15]: remove this console log please

File: frontend/src/apps/main/components/navbar/index.tsx [hunk 2]
```diff
@@ -1,19 +1,19 @@
...
-import { useState } from "react";
+import * as React from "react";
+import { useDispatch, useSelector } from "react-redux";
+import { animated, useSpring } from "react-spring";
+import { LinkV2 } from "src/components/link-v2";
+import logo from "t9/apps/main/assets/png/logo.png";
+import AppSearch from "t9/apps/main/components/navbar/app-search/index";
+import { actionType } from "t9/apps/main/redux/constants";
+import { StateInterface } from "t9/types/main";
+import IosSwitch from "./ios-switch";
```
Comment (Souhaib-Benbouzid) [inline frontend/src/apps/main/components/navbar/index.tsx:16]: run yarn lint:fix to sort this imports.

**Review by Souhaib-Benbouzid — CHANGES_REQUESTED** [review]
good work, please review the comments I left.

File: frontend/src/apps/main/components/navbar/index.tsx (hunk 2 above)
Comment (NurElHuda) [reply frontend/src/apps/main/components/navbar/index.tsx:16]: yes, I have used it. But it isn't working. It isn't fixing the sort errors. It just flags the errors but don't fix them.

File: frontend/src/apps/main/redux/reducers/articles-page/index.ts (hunk 1 above)
Comment (NurElHuda) [reply frontend/src/apps/main/redux/reducers/articles-page/index.ts]: Done.

File: frontend/src/apps/main/components/navbar/index.tsx (hunk 2 above)
Comment (NurElHuda) [reply frontend/src/apps/main/components/navbar/index.tsx]: I've fixed it.

**Comment by NurElHuda** [comment]
This pull request adds a search input component to the nav-bar. The search doesn't work just yet.
This is the first part of implementing the search functionalities.

**Comment by ZibanPirate** [comment]
Understood 👌🏽, later we will add and API endpoint for search, only then you can integrate both stacks

**Comment by ZibanPirate** [comment]
thanks again for the PR @NurElHuda , unfortunately we now switched default branch to `main` and made breaking changes, from now on, all PRs should be against `main`, so please close this PR and if you have time, create a new one against `main`

**Comment by ZibanPirate** [comment]
i'm closing this for now @NurElHuda , we can always cherry pick some commit from here when needed

**Pull request closed without merging**